app/src/main/java/uk/gov/android/securestore/authentication/UserAuthenticator.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.android.securestore.authentication
+
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_STRONG
+import androidx.biometric.BiometricManager.Authenticators.DEVICE_CREDENTIAL
+import androidx.biometric.BiometricPrompt
+import androidx.fragment.app.FragmentActivity
+import uk.gov.android.securestore.AccessControlLevel
+
+class UserAuthenticator(
+    private val context: FragmentActivity
+) : Authenticator {
+    override fun authenticate(
+        accessControlLevel: AccessControlLevel,
+        configuration: AuthenticatorPromptConfiguration,
+        handler: AuthenticatorCallbackHandler
+    ) {
+        val promptInfoBuilder = BiometricPrompt.PromptInfo.Builder()
+            .setTitle(configuration.title)
+            .setSubtitle(configuration.subTitle)
+            .setDescription(configuration.description)
+            .setAllowedAuthenticators(getRequireAuthenticators(accessControlLevel))
+
+        val biometricPrompt = BiometricPrompt(
+            context,
+            object : BiometricPrompt.AuthenticationCallback() {
+                override fun onAuthenticationError(
+                    errorCode: Int,
+                    errString: CharSequence
+                ) {
+                    super.onAuthenticationError(errorCode, errString)
+                    handler.onError(errorCode, errString)
+                }
+
+                override fun onAuthenticationSucceeded(
+                    result: BiometricPrompt.AuthenticationResult
+                ) {
+                    super.onAuthenticationSucceeded(result)
+                    handler.onSuccess()
+                }
+
+                override fun onAuthenticationFailed() {
+                    super.onAuthenticationFailed()
+                    handler.onFailure()
+                }
+            }
+        )
+
+        biometricPrompt.authenticate(
+            promptInfoBuilder.build()
+        )
+    }
+
+    private fun getRequireAuthenticators(accessControl: AccessControlLevel) =
+        when (accessControl) {
+            AccessControlLevel.OPEN -> -1
```

Comment:
What does `-1` mean here?

Reply:
-1 is just basically an no return. There is no authenticator for access level `OPEN`

Reply:
Could the -1 be constant instead of a magic number? That should make it easier to understand

Reply:
Do you want to apply AUTH_TYPE_OPEN here instead of -1?